### PR TITLE
feat: Cleanup asset accounts

### LINKED ISSUE #8845: [Feature]: Asset account refactoring to add network treasury and global insurance accounts
# Feature Overview

A refactoring of asset account is needed. To support the following:

**Network Treasury**
1 per asset
deposit to 0x0 to fund from Ethereum
can transfer into here from any normal account
can be destination of governance initiated transfer
only way to transfer out should be by a governance initiated one off or recurring transfer

**Global Insurance Pool**
1 per asset
recipient of insurance pool funds when a market closes
currently only way to transfer out should be by a governance initiated one off or recurring transfer

### PR DESCRIPTION
**Pull request by ze97286**
Closes #8845

1. Adding network treasury account to asset
2. Adding global insurance account to asset
3. transferring leftover and insurance pool into the global insurance account
4. transferring LP penalties stuff to the network treasury
5. Handle those accounts in checkpoint and snapshot
6. Allow transfer to/from these accounts in a gov transfer

### CODE REVIEW

File: datanode/gateway/graphql/schema.graphql [hunk 1]
```diff
@@ -3460,6 +3464,8 @@ enum AccountType {
   ACCOUNT_TYPE_HOLDING
   "Per liquidity provider, per market account for holding LPs' fees before distribution"
   ACCOUNT_TYPE_LP_LIQUIDITY_FEES
+  "NetworkTeasury, per asset treasury controlled by the network"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:3467]:
```suggestion
  "Network treasury, per-asset treasury controlled by the network"
```

File: core/integration/stubs/broker_stub.go [hunk 2]
```diff
@@ -720,7 +720,7 @@ func (b *BrokerStub) GetMarketInsurancePoolAccount(market string) (types.Account
 	return types.Account{}, errors.New("account does not exist")
 }
 
-func (b *BrokerStub) GetAssetNetworkTreasuryAccount(asset string) (types.Account, error) {
+func (b *BrokerStub) GetStakingRewwardAccount(asset string) (types.Account, error) {
```
Comment (candida-d) [inline core/integration/stubs/broker_stub.go:723]:
```suggestion
func (b *BrokerStub) GetStakingRewardAccount(asset string) (types.Account, error) {
```

File: core/collateral/engine.go [hunk 3]
```diff
@@ -811,9 +843,9 @@ func (e *Engine) CheckLeftOverBalance(ctx context.Context, settle *types.Account
 			Asset:       asset,
 			Type:        types.TransferTypeClearAccount,
 		}
-		globalRewardPool, _ := e.GetGlobalRewardAccount(asset)
+		globalIns, _ := e.GetNetworkTreasuryAccount(asset)
```
Comment (jeremyletang) [inline core/collateral/engine.go:846]:
```suggestion
		networkTreasury, _ := e.GetNetworkTreasuryAccount(asset)
```

File: core/collateral/engine.go [hunk 4]
```diff
@@ -811,9 +843,9 @@ func (e *Engine) CheckLeftOverBalance(ctx context.Context, settle *types.Account
...
 		req.FromAccount[0] = settle
-		req.ToAccount = []*types.Account{globalRewardPool}
+		req.ToAccount = []*types.Account{globalIns}
```
Comment (jeremyletang) [inline core/collateral/engine.go:848]:
```suggestion
		req.ToAccount = []*types.Account{networkTreasury}
```

File: core/collateral/engine.go [hunk 5]
```diff
@@ -277,6 +277,38 @@ func (e *Engine) EnableAsset(ctx context.Context, asset types.Asset) error {
...
+			Balance:  num.UintZero(),
+			MarketID: noMarket,
+			Type:     types.AccountTypeNetworkTreasury,
+		}
+		e.accs[netTreasury] = ntAcc
+		e.addAccountToHashableSlice(ntAcc)
+		e.broker.Send(events.NewAccountEvent(ctx, *ntAcc))
+	}
+
+	// network treasury for the asset
```
Comment (jeremyletang) [inline core/collateral/engine.go:296]:
```suggestion
	// global insurance pool for the asset
```

**Review by jeremyletang — DISMISSED** [review]

File: core/collateral/engine.go (hunk 3 above)
Comment (ze97286) [reply core/collateral/engine.go]: going with global insurance pool

**Review by candida-d — APPROVED** [review]

**Review by jeremyletang — APPROVED** [review]

**Comment by ze97286** [comment]
> A generic question: will we need to migrate any balances when this goes onto mainnet?

I don't believe so. Any balance that currently goes towards the "global reward account" is being distributed as staking reward. Worst case if funds remain in the global reward account they can be moved using the governance transfer.

File: core/integration/features/verified/Funding-treasury.feature [hunk 6]
```diff
@@ -220,7 +220,7 @@ Feature: check the insurance pool getting shared equally between all markets wit
       | party | reference | status         |
       | aux   | aux-1-19  | STATUS_STOPPED |
 
-    And the network treasury balance should be "1824" for the asset "USD"
+    And the global insurance pool balance should be "1824" for the asset "USD"
```
Comment (davidsiska-vega) [inline core/integration/features/verified/Funding-treasury.feature:223]: On line 217 above, can we make the comment say "global insurance pool"?

**Review by davidsiska-vega — APPROVED** [review]
Feature tests look fine, apart from the comments in the feature tests referring to "network treasury" instead of "global insurance pool".

File: core/integration/features/verified/Funding-treasury.feature (hunk 6 above)
Comment (ze97286) [reply core/integration/features/verified/Funding-treasury.feature:223]: done

**Pull request merged**